core/node/api_server/src/execution_sandbox/apply.rs

```diff
@@ -30,18 +23,15 @@ use zksync_system_constants::{
 };
 use zksync_types::{
     api,
-    block::{pack_block_info, unpack_block_info, L2BlockHasher},
+    block::{unpack_block_info, L2BlockHasher},
     fee_model::BatchFeeInput,
-    get_nonce_key,
-    utils::{decompose_full_nonce, nonces_to_full_nonce, storage_key_for_eth_balance},
-    AccountTreeId, L1BatchNumber, L2BlockNumber, Nonce, ProtocolVersionId, StorageKey, Transaction,
-    H256, U256,
+    AccountTreeId, L1BatchNumber, L2BlockNumber, ProtocolVersionId, StorageKey, H256, U256,
 };
-use zksync_utils::{h256_to_u256, time::seconds_since_epoch, u256_to_h256};
+use zksync_utils::{h256_to_u256, time::seconds_since_epoch};
 
 use super::{
-    vm_metrics::{self, SandboxStage, SANDBOX_METRICS},
-    ApiTracer, BlockArgs, OneshotExecutor, TxExecutionArgs, TxSetupArgs,
+    vm_metrics::{SandboxStage, SANDBOX_METRICS},
+    BlockArgs, TxSetupArgs,
 };
 
 pub(super) async fn prepare_env_and_storage(
```

Comment:
With some creativity, `BlockArgs`, `TxSetupArgs` and this logic can be (mostly) moved to the executor crate. Didn't do that to keep the PR size reasonable, and because it's not *completely* trivial; e.g., I'd like to isolate the moved code from knowing that `BlockArgs` are obtained from `api::BlockId` etc., so that oneshot executor isn't 100% tied to the API server domain.